### DISCUSSION TITLE
LogLogistic distribution for SPEI

### DISCUSSION THREAD

**Discussion by lmuntjewerf** [monocongo/climate_indices#482] [Ideas]
Hi!

So I am in a project where I need to calculate SPEI. Python is my preferred language, so I came across your package. It works perfectly. 

Now, from literature my understanding is that the log-logistic distribution is recommended for SPEI (refs 1,2). The climate indices checkout I have can do gamma and Pearson distribution. I am thinking to adapt my local code to accommodate the log-logistic distribution. It can be done with scipy.stats.fisk (fisk distribution is also known as the log-logistic distribution.)

Before I get into this, I was wondering if there is any work on you side being done on this. 
Looking forward to hear from you! 

Thanks, 
Laura 

refs
1. Vicente-Serrano, S. M., Beguería, S., & López-Moreno, J. I. (2010). A multiscalar drought index sensitive to global warming: the standardized precipitation evapotranspiration index. Journal of climate, 23(7), 1696-1718.
2. Vicente‐Serrano, S. M., & Beguería, S. (2016). Comment on ‘Candidate distributions for climatological drought indices (SPI and SPEI)’by James H. Stagge et al. International Journal of Climatology, 36(4), 2120-2131.

**Comment by monocongo** [comment]
Thanks so much for your participation, Laura. This would be a very welcome contribution to this project. As a matter of fact, I had the same idea several years back, see issue #106  No active development is underway on that issue as far as I can tell. Please, have at it! When/if you get stuck please don't hesitate to reach out, and thanks in advance!

**Reply by lmuntjewerf** [reply]
Thanks for getting back to me so quickly! Ok, perfect I will try and implement it. Definitely I will let you know how that goes or when I hit problems.

**Comment by AhhyeonC** [comment]
I think this is very good idea. I always have thought that this package is very useful, but limited type of distribution function is also a drawback of this package as well. I should also try scipy.stats.fisk. Thank you for your idea.

**Comment by lmuntjewerf** [comment]
Hi @monocongo 

I've read up on the previous work in #106 and the more recent issue #440.  I reached out to @Emmadd who did the work in #440. My understanding now is their fork code is almost ready to merge into main, it needs tests to add to the testing scripts. 

 I can work on adding the testing. What do you think?

**Comment by monocongo** [comment]
Yes, absolutely, please have a go at it. My apologies in advance for not having too much time now to devote to the review and testing of a new implementation, but if you follow the same general patterns in place and document as much as possible then you're spoon-feeding me and I will do my best. I appreciate your contributions (and @Emmadd!), this will be a very nice addition.

**Comment by monocongo** [comment]
I have discovered that the good folks at [xclim](https://github.com/Ouranosinc/xclim) have implemented [support for log-logistic distribution](https://github.com/Ouranosinc/xclim/pull/1183) in their package, we might be able to use the same approach or somehow use their code as-is so as to not reinvent this wheel?
